test/conformance/conformance_diff_test.go

```diff
@@ -0,0 +1,131 @@
+package conformance
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/bom-squad/protobom/pkg/formats"
+	"github.com/bom-squad/protobom/pkg/reader"
+	"github.com/bom-squad/protobom/pkg/sbom"
+	"github.com/bom-squad/protobom/pkg/writer"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+)
+
+type fakeBuf struct {
+	*bufio.ReadWriter
+}
+
+func (fwc *fakeBuf) Close() error {
+	return nil
+}
+
+func (f *fakeBuf) Seek(offset int64, whence int) (int64, error) {
+	return 0, nil
+}
+
+func TestDiffTranslationFormats(t *testing.T) {
+	from := formats.SPDX23JSON
+	to := formats.CDX14JSON
+
+	fromFiles := findFiles(t, from)
+	require.NotEqual(t, len(fromFiles), 0)
+
+	fname := fromFiles[0]
+
+	r := reader.New()
+	fromSut, err := r.ParseFile(fname)
+
+	require.NoError(t, err)
+	w := writer.New()
+
+	var b bytes.Buffer
+	fakeWriter := bufio.NewWriter(&b)
+	fakeReader := bufio.NewReader(&b)
+	fakeReadWriter := bufio.NewReadWriter(fakeReader, fakeWriter)
+
+	fake := fakeBuf{
+		ReadWriter: fakeReadWriter,
+	}
+
+	w.WriteStreamWithOptions(
+		fromSut, &fake, &writer.Options{Format: to},
+	)
+	toSut, err := r.ParseStreamWithOptions(&fake, &reader.Options{Format: to})
+	require.NoError(t, err)
+
+	golden := readProtobom(t, fname+".proto")
+	t.Logf("to-sut: %s golden: %s", fname, fname+".proto")
+	t.Run(
+		fmt.Sprintf("From: testDiffNodes-%s-%s-%s-TO-%s-%s-%s", from.Type(), from.Version(), from.Encoding(), to.Type(), to.Version(), to.Encoding()),
+		func(t *testing.T) {
+			testNodes(t, golden, toSut)
+			testDiffNodeList(t, golden, toSut)
+			testEdges(t, golden, toSut)
+			testDocumentDiff(t, golden, toSut)
+		},
+	)
+}
+
+func testDiffNodeList(t *testing.T, golden, sut *sbom.Document) {
+	// Compare the nodes
+	nl := golden.NodeList
+	nl2 := sut.NodeList
+
+	if len(nl.Edges) != len(nl2.Edges) {
+		require.Equal(t, len(nl.Edges), len(nl2.Edges), "edges amount differs")
+	}
+
+	if len(nl.Nodes) != len(nl2.Nodes) {
+		require.Equal(t, len(nl.Nodes), len(nl2.Nodes), "node amount differs")
+	}
+
+	if len(nl.RootElements) != len(nl2.RootElements) {
+		require.Equal(t, len(nl.RootElements), len(nl2.RootElements), "root amount differs")
+	}
+
+	r1 := nl.RootElements
+	r2 := nl2.RootElements
+	sort.Strings(r1)
+	sort.Strings(r2)
```

Comment:
You are on the right track here, but these expanded tests should augment the ones we have in the conformance tests which are very simple right now:

https://github.com/bom-squad/protobom/blob/c2d38e911b0b468130d2b6257e219ff84ef7d689/test/conformance/conformance_test.go#L60-L63

We should start adding more deeper test cases. Right now what we do is simply test that the documents more or less "look the same", that is have the same number of nodes and edges, etc. But we should add more comprehensive comparisons.

Also remember that we already have diff support in the protobom API, so avoid duplicating that.

Reply:
Yeah i tried to use the simple string compare failed to match when running the test .. Let me review the diff probably missed it..

Reply:
I decided to expand the current conformance on a separate PR so this one can stay about diff between the source and translated version of the same document.

Take a peek in https://github.com/bom-squad/protobom/pull/163 see if correlates.